GameDesign/Assets/Scripts/MonoBehaviours/Movements/SpawnerWaypoint.cs

```diff
@@ -0,0 +1,75 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SpawnerWaypoint : Waypoint
+{
+    [Header("Spawner properties")]
+    [SerializeField] float spawnArea;
+
+    private LevelSettingManager levelManagerSingleton;
+
+    void OnDrawGizmos()
+    {
+        Gizmos.color = sphereColor;
+        Gizmos.DrawWireSphere(transform.position, spawnArea );
+        Gizmos.color = connectingColor;
+        foreach (Waypoint wp in possibleNextPoint)
+        {
+            if (wp != null) Gizmos.DrawLine(transform.position, wp.transform.position);
+        }
+    }
+
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        levelManagerSingleton = LevelSettingManager.Instance;
+        
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        
+    }
+
+    public override void waypointReached(NPCMovementManager npcMover)
+    {
+        Debug.LogWarning("An agent had its waypoint target as a spawner. Making him go back");
+        npcMover.targetWayPoint = npcMover.previousWayPoint;
+        npcMover.agent.SetDestination(npcMover.targetWayPoint.transform.position);
+    }

```

Comment:
maybe to avoid it going back to the spawner it as previous point. the waypoint in general try not to go to the previous point.

Reply:
can you elaborate on this? I didn't understand what you meant